### PR TITLE
[WIP] Bump all eventing manifests to 1.13

### PR DESCRIPTION
**Pull request by creydr**
As per title

### CODE REVIEW

**Comment by creydr** [comment]
operator tests are failing in `TestChannelBasedBrokerToKsvc` due to

```
{"level":"info","ts":"2024-02-28T13:11:45.768Z","logger":"eventshub.event logger","caller":"logger_vent/logger.go:26","msg":"Event: \n-- EventInfo --\n--- Kind: Sent ---\n--- Error ---\nPost \"http://sink-bilgqwrz-oprkezfu.test-gdmwfrkp.svc.cluster.local\": net/http: HTTP/1.x transport connection broken: http: ContentLength=7 with Body length 0\n--- HTTP headers ---\n  Ce-Id: full-event\n  Prefer: reply\n  Ce-Knativearrivaltime: 2024-02-28T13:11:45.73522362Z\n  K-Proxy-Request: activator\n  X-Request-Id: 5bb3c8f6-4cc3-418a-a2ed-21f0d58217a1\n  Ce-Type: com.example.FullEvent\n  Ce-Exint: 42\n  Ce-Exurl: http://example.com/source\n  Content-Length: 7\n  Kn-Namespace: test-gdmwfrkp\n  Ce-Exbinary: AAECAw==\n  Ce-Time: 2020-03-21T12:34:56.78Z\n  Ce-Specversion: 1.0\n  Ce-Extime: 2020-03-21T12:34:56.78Z\n  Traceparent: 00-c6fcc8c8fdfd4798a7f330ae9f8e5292-8482c7e37108ff1f-00\n  User-Agent: Go-http-client/1.1\n  Ce-Exbool: true\n  X-Forwarded-For: 10.129.2.45, 10.129.2.43\n  Ce-Subject: topic\n  X-Forwarded-Proto: http\n  Content-Type: text/json\n  Forwarded: for=10.129.2.45;proto=http\n  Ce-Exstring: exstring\n  Ce-Source: http://example.com/source\n  Accept-Encoding: gzip\n  Ce-Dataschema: http://example.com/schema\n\n--- Origin: 'sink-bilgqwrz' ---\n--- Observer: 'sink-bilgqwrz' ---\n--- Time: 2024-02-28 13:11:45.768187899 +0000 UTC m=+18.600809202 ---\n--- Sequence: 0 ---\n--- Sent Id:  'full-event ---\n--------------------\n"}
```
-> `net/http: HTTP/1.x transport connection broken: http: ContentLength=7 with Body length 0`
in the forwarder

and then 
```
{"level":"info","ts":"2024-02-28T13:11:45.779Z","logger":"eventshub.event logger","caller":"logger_vent/logger.go:26","msg":"Event: \n-- EventInfo --\n--- Kind: Received ---\n--- Error ---\nunexpected EOF\n--- HTTP headers ---\n  User-Agent: Go-http-client/1.1\n  Traceparent: 00-c6fcc8c8fdfd4798a7f330ae9f8e5292-bfa08eef99e87a04-00\n  Prefer: reply\n  Kn-Namespace: test-gdmwfrkp\n  Content-Type: text/json\n  Host: sink-bilgqwrz.test-gdmwfrkp.svc.cluster.local\n  X-Forwarded-For: 10.129.2.45, 10.129.2.43\n  X-Forwarded-Proto: http\n  Forwarded: for=10.129.2.45;proto=http\n  X-Request-Id: 5bb3c8f6-4cc3-418a-a2ed-21f0d58217a1\n  Accept-Encoding: gzip\n  Content-Length: 7\n  K-Proxy-Request: activator\n\n--- Origin: '10.131.0.60:50246' ---\n--- Observer: 'sink-bilgqwrz' ---\n--- Time: 2024-02-28 13:11:45.768373202 +0000 UTC m=+30.121043946 ---\n--- Sequence: 1 ---\n--- Sent Id:  ' ---\n--------------------\n"}
```
-> `unexpected EOF`
in the sink

Only a very vague guess from my side, but could it be, because we read the request body via https://github.com/knative-extensions/reconciler-test/blob/acb1ce90370f6c5e120b5663d12c6c53e0e04026/pkg/eventshub/forwarder/forwarder.go#L129 already and then sending the same request in https://github.com/knative-extensions/reconciler-test/blob/acb1ce90370f6c5e120b5663d12c6c53e0e04026/pkg/eventshub/forwarder/forwarder.go#L183 again?

@mgencur as you have done a few things with the forwarder: does this ring any bells on you?

**Comment by mgencur** [comment]
> @mgencur as you have done a few things with the forwarder: does this ring any bells on you?

Unfortunately, I have no idea. But I'd probably try removing the forwarder and keep just a simple Kube service, and see if this still happens. It can be done by removing `eventshub.WithKnativeServiceForwarder` on [this line](https://github.com/openshift-knative/serverless-operator/blob/main/test/eventinge2erekt/main_test.go#L48C3-L48C40)

**Comment by creydr** [comment]
Seems https://github.com/openshift-knative/serverless-operator/pull/2527#issuecomment-1969003629 is the issue. I built an eventshub image which has https://github.com/knative-extensions/reconciler-test/commit/a7237b013cbb4565dc9a7bc8c15df0866295fe23 reverted, used it in the images.yaml and then the eventinge2erekt tests passed.
Let me temporarily push this to see if the other operator tests pass too.

**Comment by creydr** [comment]
Summary for the failing `operator-e2e` so far:
* Failures seems to be related to the forwarder (and https://github.com/knative-extensions/reconciler-test/commit/a7237b013cbb4565dc9a7bc8c15df0866295fe23)
* temporarily I switched the eventshub image for the e2e tests to a build including the fix from https://github.com/knative-extensions/reconciler-test/pull/681 (see https://github.com/openshift-knative/serverless-operator/pull/2527/commits/34696b512919cf9fc5c54107b207bc7deadb7461). This was tested "locally" already successfully

File: test/images-rekt.yaml [hunk 1]
```diff
@@ -1,2 +1,3 @@
-knative.dev/reconciler-test/cmd/eventshub: quay.io/openshift-knative/eventing/eventshub:v1.13
+#knative.dev/reconciler-test/cmd/eventshub: quay.io/openshift-knative/eventing/eventshub:v1.13
+knative.dev/reconciler-test/cmd/eventshub: quay.io/creydr/knative/eventing/eventshub:write-event-to-request-again
```
Comment (matzew) [inline test/images-rekt.yaml:2]: :+1:

**Comment by creydr** [comment]
operator tests passed in https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-knative_serverless-operator/2527/pull-ci-openshift-knative-serverless-operator-main-414-operator-e2e-aws-414/1762870803679940608.
@matzew @mgencur @pierDipi could you PTAL at https://github.com/knative-extensions/reconciler-test/pull/681 then?

**Comment by creydr** [comment]
Upgrade tests are failing (at least due to):
```
Version migration is not eligible with message: not supported to upgrade or downgrade across multiple MINOR versions. 
```
So removing this check, as discussed in https://redhat-internal.slack.com/archives/CEXRYS5QC/p1708945005582059?thread_ts=1708599613.277839&cid=CEXRYS5QC and https://redhat-internal.slack.com/archives/CEXRYS5QC/p1709116187511639?thread_ts=1709044877.485049&cid=CEXRYS5QC:
06d076bef0479b128648e2bd762f7ac7d6c1a256

**Comment by creydr** [comment]
The upgrade tests are failing, because we test to downgrade and SO v1.32 still have the version check to not skip minor versions. 

```
FAIL test/upgrade.TestServerlessUpgradePrePost/Run/Steps/DowngradeWith/DowngradeServerless
...
Version migration is not eligible with message: not supported to upgrade or downgrade across multiple MINOR versions.
```

So we need the https://github.com/openshift-knative/serverless-operator/pull/2529 (which is the same as 06d076bef0479b128648e2bd762f7ac7d6c1a256) to be backported to 1.32

**Comment by creydr** [comment]
`NoClusters: No clusters in pool are ready to be claimed`
/test 414-mesh-e2e-aws-414

**Comment by pierDipi** [comment]
```
2024-03-05T12:14:00.088429598Z {"level":"error","ts":"2024-03-05T12:14:00.088Z","logger":"pingsource-mt-adapter","caller":"mtping/runner.go:157","msg":"failed to send cloudevent result: {result 26 0  403: } {source 15 0 /apis/v1/namespaces/tenant-2/pingsources/pingsource-neunacfy <nil>} {target 15 0 http://sink-nrguxlwm.tenant-1.svc.cluster.local <nil>} {id 15 0 cf3632b0-731a-438c-99f1-311be318b1bc <nil>}","commit":"15e55a4-dirty","knative.dev/pod":"pingsource-mt-adapter-5cbddcf566-7gjd4","stacktrace":"knative.dev/eventing/pkg/adapter/mtping.(*cronJobsRunner).cronTick.func1\n\t/go/src/github.com/openshift/origin/pkg/adapter/mtping/runner.go:157\ngithub.com/robfig/cron/v3.FuncJob.Run\n\t/go/src/github.com/openshift/origin/vendor/github.com/robfig/cron/v3/cron.go:136\ngithub.com/robfig/cron/v3.(*Cron).startJob.func1\n\t/go/src/github.com/openshift/origin/vendor/github.com/robfig/cron/v3/cron.go:312"}
```

**Comment by pierDipi** [comment]
That error is expected as `TestPingSourceToKsvcCrossTenant`is passing

**Comment by creydr** [comment]
Closing as this was the test PR. Bump done in #2567 after we had the branch rename (1.13 -> 1.12)

**Pull request closed without merging**